File: client/GhostClient/Core/Network/Packets/StatsUpdatePacket.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using GhostClient.Core;

namespace Ghost.Core.Network.Packets
{
    public class StatsUpdatePacket : Packet
    {
        public StatsUpdatePacket(Stream stream) : base(stream)
        {
        }

        public StatsUpdatePacket(Stream stream, bool autoClose) : base(stream, autoClose)
        {
        }

        public StatsUpdatePacket(byte[] data) : base(data)
        {
        }

        protected override void OnHandlePacket()
        {
            string id = Consume(4).AsciiStringValue;
            double value = Consume(8).DoubleValue;

            switch (id)
            {
                case "mspd":
                    GameHandler.Game.player1.SpeedStat = value;
                    break;
                case "frte":
                    GameHandler.Game.player1.FireRateStat = value;
                    break;
            }
        }
    }
}

File: client/core/src/me/eddiep/ghost/client/network/packets/StatsUpdatePacket.kt
package me.eddiep.ghost.client.network.packets

import me.eddiep.ghost.client.network.Packet
import me.eddiep.ghost.client.network.PlayerClient

class StatsUpdatePacket : Packet<PlayerClient>() {

    override fun handle() {
        val id = consume(4).asString()
        val value = consume(8).asDouble()

        when (id) {
            "mspd" -> client.game.player1.speedStat = value
            "frte" -> client.game.player1.fireRateStat = value
        }
    }
}
